enterprise/server/util/cpuset/cpuset_test.go

```diff
@@ -104,3 +104,60 @@ func TestCPUSetDisabledManualCPUSet(t *testing.T) {
 	defer cancel1()
 	assert.Equal(t, []int{0, 1, 3}, cpus1)
 }
+
+func TestNumaNodeFairness(t *testing.T) {
+	flags.Set(t, "executor.cpu_leaser.enable", true)
+	flags.Set(t, "executor.cpu_leaser.overhead", 0)
+	flags.Set(t, "executor.cpu_leaser.min_overhead", 0)
+	flags.Set(t, "executor.cpu_leaser.cpuset", "0:0-3,1:4-7")
+
+	var allCPUs []int
+
+	cs, err := cpuset.NewLeaser()
+	require.NoError(t, err)
+	task1 := uuid.New()
+	cpus1, cancel1 := cs.Acquire(4000, task1)
+	defer cancel1()
+	allCPUs = append(allCPUs, cpus1...)
+
+	task2 := uuid.New()
+	cpus2, cancel2 := cs.Acquire(4000, task2)
+	defer cancel2()
+	allCPUs = append(allCPUs, cpus2...)
+
+	assert.ElementsMatch(t, []int{0, 1, 2, 3, 4, 5, 6, 7}, allCPUs)
```

Comment:
probably worth adding a third task and checking that the CPUs are on the same node

Reply:
Done.